### PR TITLE
Better error message for an invalid profile id

### LINKED ISSUE #858: Error message when creating a Portal Profile is not explicit
While creating a portal profile and the name include a space for instance, the error message is "XName does not match", we should change it for something like "the Name you used is invalid"

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Better error message for an invalid profile id
## Impacts

The admin gui
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++
- Improve error message in the profile managment page

Bug Fixes
+++++++++
Fixes #858
## Delete branch after merge

NO

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm [hunk 1]
```diff
@@ -63,7 +63,7 @@ has_field 'id' =>
    type => 'Text',
    label => 'Profile Name',
    required => 1,
-   apply => [ { check => qr/^[a-zA-Z0-9][a-zA-Z0-9\._-]*$/ } ],
+   apply => [ { check => qr/^[a-zA-Z0-9][a-zA-Z0-9\._-]*$/, message => 'An invalid character in profile id. The profile id can only contain alphanumeric characters, dashes, period and or underscores' } ],
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm]: Line seems a bit long, can you put it on 2 lines
:nail_care:

**Comment by jrouzierinverse** [comment]
Please review

File: html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm [hunk 2]
```diff
@@ -63,7 +63,12 @@ has_field 'id' =>
    type => 'Text',
    label => 'Profile Name',
    required => 1,
-   apply => [ { check => qr/^[a-zA-Z0-9][a-zA-Z0-9\._-]*$/ } ],
+   apply => [
+       {   check => qr/^[a-zA-Z0-9][a-zA-Z0-9\._-]*$/,
+           message =>
+             'An invalid character in profile id. The profile id can only contain alphanumeric characters, dashes, period and or underscores'
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm]: Grammar is not right

Suggestions :
An invalid character was found....
Found an invalid character...

Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm]: How about this?
Invalid character(s) was found in the profile id provided.
A profile id can only contain alphanumeric characters, dashes, period and or underscores

Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm]: A shorter message
The profile id is invalid.
A profile id can only contain alphanumeric characters, dashes, period and or underscores.

Also I will update the help section of the profile creation page with the following text.

A profile id can only contain alphanumeric characters, dashes, period and or underscores.

**Comment by jrouzierinverse** [comment]
Please review

**Pull request merged**